src/Widgets/AppEntry.vala

```diff
@@ -74,23 +76,25 @@ public class Notifications.AppEntry : Gtk.ListBoxRow {
         );
         settings.bind ("expanded", expander, "expanded", DEFAULT);
 
-        update_expanded_state (expander.expanded);
-        expander.notify["expanded"].connect (() => update_expanded_state (expander.expanded));
+        update_expanded_state ();
+        expander.notify["expanded"].connect (update_expanded_state);
 
         clear_btn_entry.clicked.connect (() => {
             clear (); // Causes notification list to destroy this app entry after clearing its notification entries
         });
     }
 
-    private void update_expanded_state (bool expanded) {
+    private void update_expanded_state () {
         foreach (var notification in app_notifications) {
-            notification.revealer.reveal_child = expanded;
+            notification.revealer.reveal_child = expander.expanded;
         }
     }
 
     public void add_notification_entry (NotificationEntry entry) {
         app_notifications.prepend (entry);
         entry.clear.connect (remove_notification_entry);
+
+        entry.revealer.reveal_child = expander.expanded;
```

Comment:
```suggestion
        expander.bind_property ("expanded", entry.revealer, "reveal-child", SYNC_CREATE);
```

`GLib.Object` binds are different of `GLib.Settings` ones, they are one sided unless `GLib.BindingFlags.BIDIRECTIONAL` is specified.

that will let you drop the update_expanded_state() callback.

Reply:
Updated